processor/src/test/java/org/mapstruct/ap/test/unmappedtarget/beanmapping/BeanMappingIgnoreTargetTest.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright MapStruct Authors.
+ *
+ * Licensed under the Apache License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.mapstruct.ap.test.unmappedtarget.beanmapping;
+
+import org.mapstruct.ap.testutil.IssueKey;
+import org.mapstruct.ap.testutil.ProcessorTest;
+import org.mapstruct.ap.testutil.WithClasses;
+import org.mapstruct.ap.testutil.compilation.annotation.CompilationResult;
+import org.mapstruct.ap.testutil.compilation.annotation.Diagnostic;
+import org.mapstruct.ap.testutil.compilation.annotation.ExpectedCompilationOutcome;
+
+@IssueKey("3539")
+public class BeanMappingIgnoreTargetTest {
+
+    @ProcessorTest
+    @WithClasses({Source.class, Target.class, BeanMappingIgnoreTargetMapper.class})
+    public void shouldIgnoreTargetProperty() {
+        BeanMappingIgnoreTargetMapper.INSTANCE.convert( new Source() );
```

Comment:
Shouldn't there be an assertion? :)
```java
    public void shouldIgnoreTargetProperty() {
        Source source = new Source();
        Source.NestedSource nested = new Source.NestedSource();
        source.setAge( 18 );
        source.setName( "John" );
        source.setNested( nested );
        nested.setScore( 50.0 );
        Target convert = BeanMappingIgnoreTargetMapper.INSTANCE.convert( source );
        assertThat(convert).hasNoNullFieldsOrPropertiesExcept( "address" );
        assertThat(convert.getNested()).hasNoNullFieldsOrPropertiesExcept( "flag" );
    }
``` 

SHouldn't we add a test like that? :) Or is it enough thats it does not generate a warning that we have unmapped targets

Reply:
I agree about this one. We should add an assertion as well.